app/controllers/coronavirus_form/postcode_lookup_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::PostcodeLookupController < ApplicationController
+  def show
+    if session[:flash].present?
+      flash.now[:validation] = [{ text: session[:flash], field: "postcode" }]
+      session.delete(:flash)
+      @postcode_entered = true
+      render controller_path, status: :unprocessable_entity
+    else
+      render controller_path, status: :ok
+    end
+  end
+
+  def submit
+    redirect_to check_your_answers_url if session[:check_answers_seen]
+
+    if params[:postcode].blank?
+      flash.now[:validation] = [{ text: I18n.t("coronavirus_form.questions.address_lookup.hint"), field: "postcode" }]
+      render controller_path, status: :unprocessable_entity
+    else
+      errors = helpers.validate_postcode("postcode", params[:postcode])
```

Comment:
We tried removing the helpers calls here and it caused a few method errors in the tests.

Just had a look at this and yes it does seem a bit unnecessary to have that call here but the `validate_postcode` method already does a check for a blank postcode. It does this to stop itself falling over when it gets to the regex that checks format. It is possible to input an error if postcode is blank in the `validate_postcode` method. If we want to move this check to the `validate_postcode` method we have two options. 

One is to remove `POSTCODE` from the `REQUIRED_FIELDS` constant in the `support_address_controller`. This will stop the validation happening twice but is quite horrible because we'd kind of be saying that postcode both was and wasn't mandatory. 

The other would be to move the `validate_missing_fields` method from the `support_address_controller` into the `field_validation_helper`. This is doable but would mean moving quite a bit of stuff around. It would also make the i18n structure a bit brittle because we'd want it to be universal for all of the required fields across the form. 

I think this would be worth more thought if we started doing this in more cases. As it stands though, postcode is a bit of a snowflake because of the regex validation that we're doing in the `field_validation_helper`. Because of this I don't think it's worth the downsides / effort.